lib/mailbox-handler.js

```diff
@@ -19,99 +20,92 @@ class MailboxHandler {
                 redis: this.redis,
                 pushOnly: true
             });
+
+        this.settingsHandler = new SettingsHandler({ db: this.database });
     }
 
     create(user, path, opts, callback) {
-        this.database.collection('mailboxes').findOne(
-            {
-                user,
-                path
-            },
-            (err, mailboxData) => {
-                if (err) {
-                    return callback(err);
-                }
-                if (mailboxData) {
-                    const err = new Error('Mailbox creation failed with code MailboxAlreadyExists');
-                    err.code = 'ALREADYEXISTS';
-                    err.responseCode = 400;
-                    return callback(err, 'ALREADYEXISTS');
-                }
+        this.createAsync(user, path, opts)
+            .then(mailboxData => callback(null, ...Object.values(mailboxData)))
```

Comment:
The ordering for object keys is not always obvious, and it is better to use specific values and not depend on the ordering of keys in the method response. This seems something that might break in the future where seemingly a simple change (like adding an additional property or using Object.assign) will mess up the current key order.

Just for the reference, this is how object keys are enumerated:
> Within each component of the prototype chain, all non-negative integer keys (those that can be array indices) will be traversed first in ascending order by value, then other string keys in ascending chronological order of property creation.